src/app/ngrx/reducers/selectors/error.selectors.ts

```diff
@@ -1,20 +1,83 @@
-  }
-  if (episodeError) {
-    errors.push(`${errorType(episodeError.code)} error occurred while requesting list of episodes`);
-  }
-  return errors;
-});
-export const selectMetricsErrors = createSelector(selectPodcastMetricsError, selectEpisodeMetricsError,
-  (podcasts, episodes) => podcasts.concat(episodes));
-export const selectErrors = createSelector(selectCatalogErrors, selectMetricsErrors, (catalog, metrics) => catalog.concat(metrics));
+export const select500ErrorReloadActions =
+  createSelector(selectRouter,
+    selectPodcastError,
+    selectEpisodeError,
+    selectPodcastMetricsError,
+    selectEpisodeMetricsError,
+    selectPodcastRanksError,
+    selectPodcastTotalsError,
+  (routerParams: RouterParams,
+   podcastError: any,
+   episodeError: any,
+   podcastMetricsErrors: PodcastMetricsModel[],
+   episodeMetricsErrors: EpisodeMetricsModel[],
+   podcastRanksError: any,
+   podcastTotalsError: any) => {
+    let actions = [];
+    if (podcastError && podcastError.status === 500) {
+      actions.push(new ACTIONS.CastlePodcastPageLoadAction({page: 1, all: true}));
+    }
+    if (routerParams.metricsType === METRICSTYPE_DOWNLOADS) {
+      if (episodeError && episodeError.status === 500) {
+        actions.push(new ACTIONS.CastleEpisodePageLoadAction({podcastId: routerParams.podcastId, page: 1, all: true}));
+      }
+      if (podcastMetricsErrors && podcastMetricsErrors.length) {
+        actions = actions.concat(podcastMetricsErrors
+          .filter(m => m.error && m.error.status === 500)
+          .map(m => new ACTIONS.CastlePodcastMetricsLoadAction({
+            id: m.id,
+            metricsType: routerParams.metricsType,
+            interval: routerParams.interval,
+            beginDate: routerParams.beginDate,
+            endDate: routerParams.endDate
+          })));
+      }
+      if (episodeMetricsErrors && episodeMetricsErrors.length) {
+        actions = actions.concat(episodeMetricsErrors
+          .filter(m => m.error && m.error.status === 500)
+          .map(m => new ACTIONS.CastleEpisodeMetricsLoadAction({
+            podcastId: m.podcastId,
+            guid: m.guid,
+            page: m.page,
+            metricsType: routerParams.metricsType,
+            interval: routerParams.interval,
+            beginDate: routerParams.beginDate,
+            endDate: routerParams.endDate
+          })));
+      }
+    } else {
+      if (podcastRanksError) {
+        const { podcastId, group, interval, beginDate, endDate } = routerParams;
+        actions.push(new ACTIONS.CastlePodcastRanksLoadAction({
+          id: podcastId,
+          group,
+          interval,
+          beginDate,
+          endDate
+        }));
+      }
+      if (podcastTotalsError) {
+        const { podcastId, group, interval, beginDate, endDate } = routerParams;
+        actions.push(new ACTIONS.CastlePodcastTotalsLoadAction({
```

Comment:
Yeah, these will probably cover 99% of the castle-type timeout errors out there.